### PR TITLE
Adding with_icu option and updating to harfbuzz 2.4.0

### PR DESCRIPTION
**Pull request by Maddimax**

### CODE REVIEW

**Comment by Maddimax** [comment]
This should of course be a new branch "2.4.0"

File: conanfile.py [hunk 1]
```diff
@@ -72,6 +81,11 @@ def configure_cmake(self):
             cmake.definitions['CMAKE_POSITION_INDEPENDENT_CODE'] = self.options.fPIC
         cmake.definitions["HB_HAVE_FREETYPE"] = self.options.with_freetype
         cmake.definitions["HB_BUILD_TESTS"] = False
+        cmake.definitions["HB_HAVE_ICU"] = self.options.with_icu
+
+        if self.options.with_icu:
+            cmake.definitions["CMAKE_CXX_STANDARD"] = "17"
```
Comment (uilianries) [inline conanfile.py:85]: why not `self.settings.cppstd` ?

File: conanfile.py [hunk 2]
```diff
@@ -50,13 +53,19 @@ def source(self):
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self.source_subfolder)
 
+        if self.version == "2.4.0":
```
Comment (uilianries) [inline conanfile.py:56]: as this recipe is only for 2.4.0, I think it's not required checking the version here.

File: conanfile.py [hunk 3]
```diff
@@ -72,6 +81,11 @@ def configure_cmake(self):
             cmake.definitions['CMAKE_POSITION_INDEPENDENT_CODE'] = self.options.fPIC
```
Comment (uilianries) [inline conanfile.py]: Old stuff, could be removed on this patch as improvement.

**Pull request merged**